_posts/blog/2016-10-28-durability-performance.md

```diff
@@ -0,0 +1,136 @@
+---
+title: "Durability Performance Implications"
+date: 2016-10-28 17:00:00 +0000
+author: Keith Turner
+---
+
+## Overview
+
+Accumulo stores recently written data in a sorted in memory map.  Before data is
+added to this map, it's written to an unsorted WAL (write ahead log).  In the
+case when a Tablet Server dies, the recently written data is recovered from the
+WAL.
+
+When data is written to Accumulo the following happens :
+
+ * Client sends a batch of mutations to a tablet server
+ * Tablet server does the following :
+   * Writes mutation to Tablet Servers WAL
+   * Sync or flush WAL
+   * Adds mutations to sorted in memory maps
+   * Reports success back to client.
+
+The sync/flush step above moves data written to the WAL from memory to disk.
+Write ahead logs are stored in HDFS. HDFS supports two ways of forcing data to
+disk for an open file : `hsync` and `hflush`.  
+
+## HDFS Sync/Flush Details
+
+When `hflush` is called on a WAL, it does not ensure data is on disk.  It only
```

Comment:
s/ensure/guarantee/ ? I think that better represents the "it may or may not be on disk" notion.